module/VuFind/src/VuFind/Content/Covers/Bokinfo.php

```
@@ -0,0 +1,200 @@
+ * @link     Wiki
+ * https://vufind.org/wiki/development:plugins:content_provider_components#covers
+ */
+namespace VuFind\Content\Covers;
+
+use SimpleXMLElement;
+
+/**
+ * Summon cover content loader.
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Bokinfo extends \VuFind\Content\AbstractCover implements
+    \Laminas\Log\LoggerAwareInterface,
+    \VuFindHttp\HttpServiceAwareInterface
+{
+    use \VuFind\Log\LoggerAwareTrait;
+    use \VuFindHttp\HttpServiceAwareTrait;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->supportsIsbn = $this->cacheAllowed = true;
+    }
+
+    /**
+     * Get image URL for a particular API key and set of IDs (or false if invalid).
+     *
+     * @param string $key  API key
+     * @param string $size Size of image to load (small/medium/large)
+     * @param array  $ids  Associative array of identifiers (keys may include 'isbn'
+     * pointing to an ISBN object and 'issn' pointing to a string)
+     *
+     * @return string|bool
+     */
+    public function getUrl($key, $size, $ids)
+    {
+        if (!isset($ids['isbn'])) {
+            return false;
+        }
+        if (!isset($key)) {
+            return false;
+        }
+
+        $isbn = $ids['isbn']->get13();
+        try {
+            $client = $this->createHttpClient(
+                "https://api.bokinfo.se/book/get/$isbn"
+            );
+
+            $client->getRequest()->getHeaders()
+                  ->addHeaderLine("Ocp-Apim-Subscription-Key", $key);
+
+            try {
+                $resp= $client->send();
+                $body = $resp->getBody();
+                $url = $this->getImageUrl($body);
+            } catch (\Exception $e) {
+            }
```

Do you really want to ignore exceptions completely? Should this be a return false? Or do you not need this try..catch at all because it's already inside a different try..catch?

---

Check, removing.